represetnation_labels/useful_functions.py

```diff
@@ -0,0 +1,204 @@
+        for var, vals in variables.items():
+            if var == 'Ethnicity':
+                graph_dict[dataset][var] = multi_eth_change_format(vals,dataset)
+            elif isinstance(list(vals.values())[0],dict):
+                graph_dict[dataset][var] = nested_change_format(vals,var)
+            else:
+                graph_dict[dataset][var] = change_format(vals, var)
+
+    ref_dict = {}
+    for var,vals in reference_dict['2011 Census'].items():
+        if var == 'Ethnicity':
+            ref_dict[var] = eth_change_format(vals,'ethnicity')
+        else:
+            ref_dict[var] = change_format(vals, var)
+            
+    for var,vals in ref_dict.items():
+        ref_dict[var]['percent'] = get_percents(list(vals['values']))
+
+    for dataset, variables in graph_dict.items():
+        for var,vals in variables.items():
+            missing = {get_name(k,'missing'):[v[-1]]*(len(v) -1) for k,v in vals.items() if isinstance(v[0],int)}
+            perc_dict = {get_name(k,'percent'): get_percents(v) for k,v in vals.items() if isinstance(v[0],int)}
+            ref_pers = ref_dict[var]['percent']
+            rel_dict = {get_name(k,'reletive'): get_reletives(v,ref_pers) for k,v in perc_dict.items()}
+            std_ref_dict = {get_name(k,'reference standardised'):
+                                standardise_refs(v,ref_pers) for k,v in vals.items() if isinstance(v[0],int)}
+            vals_short = {k:v[:-1] for k,v in vals.items()}
+            desc_text = {'description text': ['this is description text for this variable'] * len(ref_pers)}
+            ref_pers = {'ref percent': ref_pers}
+            rel_text = {get_name(k,'text'):[reletivise_text(i) for i in v] for k,v in rel_dict.items()}
+            graph_dict[dataset][var]={**vals_short,
+                                      **perc_dict,
+                                      **rel_dict,**std_ref_dict,**missing,**ref_pers,**desc_text,**rel_text}
+
+    return ref_dict, graph_dict
+
+def spider_plot_source(spider_dict,addition):
+    imp_keys = ['ref percent'] + [k for k in spider_dict.keys() if addition in k]
+    test_spider = {re.sub(addition, '', k): v for k, v in spider_dict.items() if k in imp_keys}
+
+    num_vars = len(test_spider['values'])
+    theta = np.linspace(0, 2 * np.pi, num_vars, endpoint=False)
+    # rotate theta such that the first axis is at the top
+    theta += np.pi / 2
+
+    def unit_poly_verts(theta):
+        """Return vertices of polygon for subplot axes.
+        This polygon is circumscribed by a unit circle centered at (0.5, 0.5)
+        """
+        x0, y0, r = [0.5] * 3
+        verts = [(r * np.cos(t) + x0, r * np.sin(t) + y0) for t in theta]
+        return verts
+
+    def radar_patch(r, theta):
+        yt = (r + 0.01) * np.sin(theta) + 0.5
+        xt = (r + 0.01) * np.cos(theta) + 0.5
+        return xt, yt
+
+    verts = unit_poly_verts(theta)
+    x = [i[0] for i in verts]
+    y = [i[1] for i in verts]
+
```

Comment:
seems like you re write a lot of these lines in `plot_functions.py`. Maybe thats because it's not appropriate to use all of `spider_plot_source` in `plot_functions.py`, if so then I'd suggest taking a few of these bits out as their own functions so you can reuse them and not repeat so much. At the least `radar_patch` and `unit_poly_verts` can be taken out.